stata_kernel/stata_magics.py

```diff
@@ -0,0 +1,314 @@
+import argparse
+import regex
```

Comment:
What's the benefit of using `regex` over the standard `re`? From what I can see it's backwards compatible with 2.7?

Reply:
I started using `regex` for named capture groups and recursive lookahead; now I'm just more used to it. Not sure if you can get either in `re`, but the specific syntax of my regexes would not port 1:1 to `re`.

Reply:
(PS: I'm not doing anything recursive, but I am doing named capture groups, which you can apparently get with a slightly different syntax; will port)

Reply:
I wasn't necessarily saying I didn't want to use `regex`, I just didn't know. And I thought I was pushing my changes to my own branch 😕

Reply:
Jaja. Ok. It's working with `re` now anyway.